Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/appservice/file/AppServiceFileNode.java

```diff
@@ -22,30 +22,81 @@
 
 package com.microsoft.tooling.msservices.serviceexplorer.azure.appservice.file;
 
+import com.google.common.util.concurrent.SettableFuture;
 import com.microsoft.azure.toolkit.lib.appservice.file.AppServiceFile;
 import com.microsoft.azure.toolkit.lib.appservice.file.AppServiceFileService;
 import com.microsoft.tooling.msservices.components.DefaultLoader;
 import com.microsoft.tooling.msservices.serviceexplorer.AzureRefreshableNode;
 import com.microsoft.tooling.msservices.serviceexplorer.Node;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
 import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.WebAppModule;
+import lombok.extern.java.Log;
+import org.apache.commons.io.FileUtils;
 import rx.Observable;
 
 import javax.swing.*;
+import java.io.File;
+import java.io.IOException;
 import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
 
+@Log
 public class AppServiceFileNode extends AzureRefreshableNode {
     private static final String MODULE_ID = WebAppModule.class.getName();
+    private static final String ERROR_DOWNLOADING = "Failed to download file[%s] to [%s].";
+    private static final String SUCCESS_DOWNLOADING = "File[%s] is successfully downloaded to [%s].";
     private final AppServiceFileService fileService;
     private final AppServiceFile file;
 
     public AppServiceFileNode(final AppServiceFile file, final Node parent, AppServiceFileService service) {
         super(file.getName(), file.getName(), parent, null);
         this.file = file;
         this.fileService = service;
+        if (this.file.getType() != AppServiceFile.Type.DIRECTORY) {
```

Comment:
override `loadActions()` and create actions there?

Reply:
`loadActions` is called before `file` is set